module/modmanager/manager.go

```diff
@@ -1082,18 +1083,19 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 
 // FirstRun is runs a module-specific setup script.
 func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
-	logger := mgr.logger.AsZap().With("name", conf.Name)
+	logger := mgr.logger.Sublogger("first_run").WithFields("module", conf.Name)
 
 	// Evaluate the Module's FirstRun path. If there is an error we assume
 	// that the first run script does not exist and we debug log and exit quietly.
 	firstRunPath, markSuccess, err := conf.EvaluateFirstRunPath(packages.LocalPackagesDir(mgr.packagesDir))
 	if err != nil {
 		// TODO(RSDK-9067): some first run path evaluation errors should be promoted to WARN logs.
-		logger.Debug("no first run script detected, skipping setup phase", "error", err)
+		logger.Debugw("no first run script detected, skipping setup phase", "error", err)
 		return nil
 	}
 
-	logger.Info("executing first run script")
+	logger = logger.WithFields("module", conf.Name, "path", firstRunPath)
```

Comment:
apparently I need to declare the full set of fields I wish to include with this logger's logs. if I only put `path` here then `module` no longer shows up, even thought I declared in a `WithFields` call earlier.

Reply:
Sorry for that rough edge -- I can confirm the code does do what you're describing.

Reply:
No worries, it's a perfectly reasonable behavior - it didn't take me long to figure out what was happening.